### PR TITLE
Consistent use of destructuring props

### LINKED ISSUE #1921: Consistent use of destructuring props
**Is your feature request related to a problem? Please describe.**
Currently there is a mixed usage of destructuring when it comes to props passed to styled components.
https://github.com/bbc/psammead/blob/3ae1fadb466a54dc0f5228f624d7ac6a8fa58fd6/packages/components/psammead-story-promo/src/index.jsx#L162-L168

**Describe the solution you'd like**
Desctrucuture `props` where we have `props.script` in `psammead-packages`

**Describe alternatives you've considered**
Leave them as they are.

**Testing notes**
[Tester to complete]

Dev insight: Regression

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #1921

**Overall change:** 
Destructure props passed to styled components consistenly accross Psammead packages.

**Code changes:**
- Desctrucuture props where we have `props.script`
   - `psammead-caption`
   - `psammead-consent-banner`
   - `psammead-headings`
   - `psammead-navigation`
   - `psammead-paragraph`
   - `psammead-story-promo`
   - `psammead-timestamp`

- Update snapshots
- Bump versions

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by andrew-nowak** [comment]
Adding a bunch of other pod labels because this isn't just front page components

File: packages/components/psammead-caption/src/index.jsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ import { scriptPropType } from '@bbc/gel-foundations/prop-types';
 import { getSansRegularItalic } from '@bbc/psammead-styles/font-styles';
 
 const Caption = styled.figcaption`
-  ${props => (props.script ? getLongPrimer(props.script) : '')};
-  color: ${C_METAL};
+  ${({ script }) => script && getLongPrimer(script)};
   ${({ service }) => getSansRegularItalic(service)}
```
Comment (andrew-nowak) [inline packages/components/psammead-caption/src/index.jsx:21]: Are we confident that these don't crash in the same way as the get type size functions? should these be `service && getX(service)`?

**Review by jroebu14 — APPROVED** [review]
Looking at the snapshots I'm fairly confident the front end won't have changed 👍

**Review by dr3 — APPROVED** [review]

**Review by andrew-nowak — COMMENTED** [review]
looks good, just a question that's possibly out of scope

File: packages/components/psammead-timestamp/src/index.jsx [hunk 2]
```diff
@@ -18,12 +18,13 @@ const PADDING = `
 `;
 
 const StyledTimestamp = styled.time`
-  ${props => (props.typographyFunc ? props.typographyFunc(props.script) : '')}
+  ${({ script, typographyFunc }) =>
+    script && (typographyFunc && typographyFunc(script))};
```
Comment (andrew-nowak) [inline packages/components/psammead-timestamp/src/index.jsx]:
```suggestion
    script && (typographyFunc && typographyFunc(script))}
```
Apparently this is causing linting to fail

**Comment by staylos92** [comment]
LGTM :+1:

**Pull request merged**